src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+import { createContext, Dispatch, ReactElement, SetStateAction, useCallback, useContext, useState } from 'react';
+
+import { CloseIcon, EditIcon } from '../../../components/icons';
+import { cx } from '../../../lib/cx';
+import { Author, ReferenceItem } from '../../../types/ReferenceItem';
+
+interface IEditingContext {
+  editing: boolean;
+  setEditing: Dispatch<SetStateAction<boolean>>;
+}
+
+const EditingContext = createContext<IEditingContext>({
+  editing: false,
+  setEditing: () => {
+    console.log('seting edtiting');
+  },
+});
+
+const returnFormatedAuthorsString = (authorsArray: Author[]): string => {
+  let authorsString = '';
+  authorsArray.forEach((a, index) => {
+    authorsString += (index > 0 ? ', ' : '') + a.fullName;
+  });
+  return authorsString;
+};
+
+const getContentString = (content: string | Author[] | undefined, id: string): string => {
+  if (typeof content == 'undefined') {
+    return '';
```

Comment:
You can define this function like this `const getContentString = (id: string, content?: string | Author[])` so you don't have to write `| undefined`